src/lib/resources/app/Installer.tsx

```diff
@@ -91,11 +94,29 @@ export class AppInstaller<TFlagName extends AvailableFlagName> {
       this.defaultFlagValues,
     );
 
-    const appVersion: AppAppVersion = await normalizeToAppVersionUuid(
-      apiClient,
-      "version" in flags ? (flags.version as string) : "latest",
-      process,
-      this.appId,
+    var appVersionId: string =
+      "version" in flags ? (flags.version as string) : "latest";
+    var appVersion: AppAppVersion;
+
+    if (validateUuid(appVersionId)) {
+      appVersion = await getAppVersionFromUuid(
+        apiClient,
+        this.appId,
+        appVersionId,
+      );
+    } else {
+      appVersion = await normalizeToAppVersionUuid(
```

Comment:
Maybe build the "if it looks like a UUID, just use that"-logic directly into `normalizeToAppVersionUuid`? IIRC, all other `normalize<whatever>` functions that we have floating around work more or less similar, anyway.